### PR TITLE
Added caddy config including 404 error page

### PR DESCRIPTION
**Pull request by tboerger**

### CODE REVIEW

File: docker/caddy.conf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+:80 {
+  root /srv/www
+
+  errors {
+    404 /404.html
```
Comment (lunny) [inline docker/caddy.conf:5]: where is the 404.html?

Comment (metalmatze) [reply docker/caddy.conf:5]: In the theme.
https://github.com/go-gitea/theme/blob/master/layouts/404.html

**Comment by metalmatze** [comment]
LGTM

**Comment by thibaultmeyer** [comment]
LGTM

**Pull request merged**